Server-side/src/auth/auth.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { RegAuthDto } from './dto/reg-auth.dto';
+import { LoginAuthDto } from './dto/login-auth.dto';
+import * as bcrypt from 'bcrypt';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    @InjectModel('user') private usersModel,
```

Comment:
Add type for usersModel 
look the example here 
https://docs.nestjs.com/techniques/mongodb